### ISSUE TITLE
Events do not get transmitted on AWS EC2 with Load Balancer

### ISSUE THREAD

**Issue by ferdinandfrank** [tlaverdure/laravel-echo-server#304]
I can't get the Laravel Echo Server to work. Everything works fine in local development mode, but when I try to use the server in production, no events get transmitted.
I'm hosting on a EC2 instance on AWS by using a load balancer with a SSL certificate.
So, the load balancer receives a HTTPS request on port 443 and redirects it to the instance via HTTP on port 80. The SSL certificate is managed on the load balancer.

How do I have to setup the `laravel-echo-server.json` file to get it to work properly? If I set it up to use HTTPS it wants me to add a path to the certificate. But the certificate is managed on the load balancer and not on the instance. If I set it up via HTTP, it does not receive any events, but it's listening with the following output:
```
- Running at localhost on port 6001
- Channels are ready.
- Listening for http events...
- Listening for redis events...
Server ready!
```

The browser's console just outputs the following over and over ([XXX] is my domain):
```
GET https://[XXX]:6001/socket.io/?EIO=3&transport=polling&t=MFGE0HQ 0 ()
```

**Comment by catalinux** [comment]
The easiest way  to do it is to manage SSL from the layer in front of the laravel-echo-server. 

You need to check your proxying settings. In nginx I have something like this

```
location /ws/{
    proxy_pass http://127.0.0.1:6001/;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
    proxy_set_header X-Forwarded-For $remote_addr;
}

**Comment by catalinux** [comment]
I confirm what @sleimanx2 ( i missed that). Your console log says it tries to connect to  6001 port. That port should not be on the client site. What is your config for the client-socket?

**Comment by catalinux** [comment]
in order to debug in browser socket.io messaging you should write this in console: 

`localStorage.debug = '*';`

then refresh

**Comment by ferdinandfrank** [comment]
Ah, nice. But I don't get any smarter with the output. Do you guys have a clue?

![socket io](https://user-images.githubusercontent.com/19409640/40984432-33c8908e-68e2-11e8-9a9a-329f05f1ee84.jpg)

**Comment by catalinux** [comment]
You're still trying to https over port 6001. You must do one of the following
- you either try to connect on https 443 - trough  balancer
- or you should connect to http over port 6001 (if that port is open to public)

**Comment by ferdinandfrank** [comment]
Thanks for your help, but I'm facing the following problem:
I can't do option one, because the port is already in use on the load balancer, so I'm receiving the following error when starting the server:
```
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE :::443
```
And when I set the port forwarding from 443 to 6001 (and not from 443 to 80), I can not access my application anymore.

For option two: I'm serving my application on https, so I get a `Mixed Content` error in the browser's console:
```
This request has been blocked; the content must be served over HTTPS.
```

**Comment by ferdinandfrank** [comment]
Ok, I was finally able to get it to work:
1. On client side, I setup a subdomain for the Echo host and set the port to 443 to use with SSL
2. On the Application Load Balancer, I setup a specific redirection rule that redirects the request from the ALB port 443 to the instance port 6001, when the host is the Echo subdomain, otherwise it just redirects to the instance port 80
3. On the Echo server I listen to port 6001 and am able to listen to all events 😃

**Comment by samuelterra22** [comment]
@ferdinandfrank Can you tell us how your final nginx configuration got?

**Comment by ferdinandfrank** [comment]
@samuelterra22 I didn't configure anything in my nginx configuration. I only setup the redirection rules on the Application Load Balancer on AWS.

**Comment by samuelterra22** [comment]
@ferdinandfrank Got it. I'm having a problem the backend is not sending the information to laravel-echo-server. The frontend connects normally but nothing is received. The backend is sending notifications normally. I'm going to open an issue.

Laravel echo server console:
![image](https://user-images.githubusercontent.com/11915449/49155418-0722b780-f302-11e8-8b78-8a56ac8578fe.png)

Backend without errors:
![image](https://user-images.githubusercontent.com/11915449/49155500-3fc29100-f302-11e8-8006-778336cbaa4f.png)

**Comment by ferdinandfrank** [comment]
@anouarabdsslm Sorry, yes it was a bit confusing. I added another step to make it more clear:
> Ok, I was finally able to get it to work (let's assume my main domain is `test.com` :
> 
> 1. On AWS Route 53 (should work on any registrar), I setup a subdomain for the Echo host `echo.test.com` which is an alias  (type `A`) that targets my Application Load Balancer (ALB)
> 2. On client side, I use the registered subdomain for the Echo host and set the port to 443 to use with SSL:
> 
> ```js
> import Echo from "laravel-echo";
> 
> window.io = require('socket.io-client');
> 
> // Have this in case we stop running our Laravel Echo server
> if (typeof io !== 'undefined') {
>     window.Echo = new Echo({
>         broadcaster: 'socket.io',
>         host: 'echo.test.com:443',
>     });
> }
> ```
> 
> 3. On the Application Load Balancer, I setup a specific redirection rule that redirects the request from the ALB port 443 to the instance port 6001, when the host is the Echo subdomain, otherwise it just redirects to the instance port 80
> 4. On the Echo server I listen to port 6001 and I am able to listen to all events 😃

**Comment by anouarabdsslm** [comment]
@ferdinandfrank thanks for your response, I end up doing something quite similar I published my solution in this article https://t.co/uV98vc8WSj

**Comment by AugmentBLU** [comment]
Hi folks, I can't seem to get my sockets to authenticate users.

I have my app and echo server on different forge instances and I can get the app to kind of talk to the echo server, however I cannot authenticate users to join presence channels. I have debugged the middleware and the user is being passed through the request but will not touch any of the routes to debug inside there.

```
Broadcast::channel('online.check', function ($user) use($controller) {
      \Log::info('route hit');
});
```
Checking the echo server output I see
`Unable to join channel. Member data for presence channel missing`

I'm not sure why as the middleware shows the user is in the request. I have Echo and Redis on one server, sessions etc are being stored in Redis, Echo is starting as expected and can see in the browser that it has connected but it will not join the presence channel.

Any ideas why? 

@anouarabdsslm I dropped a message on the post you made, I have a very similar setup but as I say, I cannot join presence channels.
